tests/test_saltworker.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+"""Salt worker main test module."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals, with_statement)
+
+import pytest
+
+from watchmaker.exceptions import InvalidValue
+from watchmaker.workers.salt import SaltBase
+
+
+@pytest.fixture
+def saltworker_client():
+    """Return salt worker arguments."""
+    system_params = {}
+    salt_config = {}
+
+    return SaltBase(system_params, **salt_config)
+
+
+def test_valid_environments(saltworker_client):
```

Comment:
I'd prefer to see the 4 test cases here split into separate test functions. The pytest output is much cleaner.

```suggestion
def test_valid_environments_default_values(saltworker_client):
    assertIsNone(saltworker_client.before_install())

def test_valid_environments_env_bogus(saltworker_client):
    saltworker_client.ent_env = 'bogus'
    with pytest.raises(InvalidValue):
        saltworker_client.before_install()

def test_valid_environments_env_missing(saltworker_client):
    saltworker_client.valid_envs = ['dev', 'test', 'prod']
    with pytest.raises(InvalidValue):
        saltworker_client.before_install()

def test_valid_environments_env_present(saltworker_client):
    saltworker_client.ent_env = 'dev'
    saltworker_client.valid_envs = ['dev', 'test', 'prod']
    assertIsNone(saltworker_client.before_install())
```

Reply:
Test 3 is no longer valid with the check for an empty array. 3 tests have been written.